ckanext/switzerland/helpers/validators.py

```diff
@@ -439,29 +446,36 @@ def ogdch_validate_formfield_temporals(field, schema):
     This validator is only used for form validation.
     The data is extracted form the temporals form fields and transformed
     into a form that is expected for database storage:
-    "temporals": [{"start_date": 0123456789, "end_date": 0123456789}]
+    "temporals": [{"start_date": <date as isodate>,
+    "end_date": <date as isodate>}]
     """
     def validator(key, data, errors, context):
-        extras = data.get(FORM_EXTRAS)
-        temporals = []
-        if extras:
-            temporals = get_temporals_from_form(extras)
-            for temporal in temporals:
-                if not temporal['start_date'] and temporal['end_date']:
-                    raise df.Invalid(
-                        _('A valid temporal must have both start and end date')  # noqa
-                    )
-                for value in temporal['start_date'], temporal['end_date']:
-                    if DATE_FORMAT_PATTERN.match(value) is None:
-                        errors[key].append(
-                            _('Expecting DD.MM.YYYY, got "%s"') % str(value)
-                        )
-                temporal['start_date'] = date_string_to_timestamp(temporal['start_date'])  # noqa
-                temporal['end_date'] = date_string_to_timestamp(temporal['end_date'])  # noqa
-        if temporals:
-            data[key] = json.dumps(temporals)
-        elif not _jsondata_for_key_is_set(data, key):
+        if key not in data:
             data[key] = '{}'
+        else:
+            if not data.get(key):
+                extras = data.get(FORM_EXTRAS)
+                if extras:
+                    temporals = get_temporals_from_form(extras)
+                    for temporal in temporals:
+                        if not temporal['start_date'] and temporal['end_date']:
+                            raise df.Invalid(
+                                _('A valid temporal must have both start and end date')  # noqa
+                            )
+            else:
+                temporals = data[key]
+
+            if not isinstance(temporals, list):
```

Comment:
`UnboundLocalError: local variable 'temporals' referenced before assignment`

Reply:
FYI: if I define temporals=[] at the beginning of this function and `run_test` some dcat-harvester  
I got an error ` [ckanext.harvest.model] create 254 Create validation Error: {'  junk': "The input field [(u'publishers', 0, u'label')] was not expected."}`